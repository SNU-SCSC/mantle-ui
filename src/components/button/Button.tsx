import React from "react";
import "./Button.css";
import Colors from "@/utils/colors";

interface ButtonProps {
  color?: Colors;
  accent?: boolean;
}

export default function Button({
  color,
  accent = false,
  //
  children,
  className,
  ...props
}: ButtonProps & React.ButtonHTMLAttributes<HTMLButtonElement>) {
  return (
    <div className={`MT-button-root ${className || ""}`}>
      <button
        className={`MT-button MT-color-${color || ""} ${accent ? "MT-accent" : ""}`}
        {...props}
      >
        {children}
      </button>
    </div>
  );
}
