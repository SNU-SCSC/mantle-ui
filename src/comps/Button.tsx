import React from "react";
import multiClassName from "@/utils/multiClassName";
import "./Button.css";

type ButtonProps = {
  /* props here */
} & React.ButtonHTMLAttributes<HTMLButtonElement>;

export default function Button({ className, children, ...props }: ButtonProps) {
  return (
    <button className={multiClassName("MT-button", className)} {...props}>
      {children}
    </button>
  );
}
