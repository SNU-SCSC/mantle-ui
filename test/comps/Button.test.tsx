import renderer from "react-test-renderer";
import React from "react";

import Button from "@/comps/Button";

describe("<Button /> on Click", () => {
  var clicked = false;
  const tree = renderer.create(<Button onClick={() => {clicked = true}}/>);
  const button = tree.root.findByType("button");

  test("Variable clicked becomes true", () => {
    expect(tree.toJSON()).toMatchSnapshot();

    renderer.act(() => {
      button.props.onClick();
    });

    expect(tree.toJSON()).toMatchSnapshot();
    expect(clicked).toBe(true);
  });
});
