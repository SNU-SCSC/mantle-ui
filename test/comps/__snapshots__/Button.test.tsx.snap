// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> on Click Variable clicked becomes true 1`] = `
<button
  className="MT-button"
  onClick={[Function]}
/>
`;

exports[`<Button /> on Click Variable clicked becomes true 2`] = `
<button
  className="MT-button"
  onClick={[Function]}
/>
`;
